diff --git a/node_modules/atscm/out/api.js b/node_modules/atscm/out/api.js
index 44adb49..499e236 100644
--- a/node_modules/atscm/out/api.js
+++ b/node_modules/atscm/out/api.js
@@ -10,11 +10,11 @@ exports.callScript = callScript;
 exports.createNode = createNode;
 exports.addReferences = addReferences;
 
-var _nodeclass = require("node-opcua/lib/datamodel/nodeclass");
+var _nodeclass = require("node-opcua");
 
-var _variant = require("node-opcua/lib/datamodel/variant");
+var _variant = require("node-opcua");
 
-var _opcua_status_code = require("node-opcua/lib/datamodel/opcua_status_code");
+var _opcua_status_code = require("node-opcua");
 
 var _Session = _interopRequireDefault(require("./lib/server/Session"));
 
@@ -60,18 +60,21 @@ function promisified(call) {
 
 
 async function withSession(action) {
-  const session = await _Session.default.create();
+  let session;
   let result = null;
   let error = null;
-
+  if (global.atscmSession) {
+    session = global.atscmSession;
+  } else {
+    session = await _Session.default.create();
+    global.atscmSession = session;
+  }
   try {
     result = await action(session);
   } catch (e) {
     error = e;
   }
 
-  await _Session.default.close(session);
-
   if (error) {
     throw error;
   }
diff --git a/node_modules/atscm/out/lib/model/Node.js b/node_modules/atscm/out/lib/model/Node.js
index b7d4333..04f45e6 100644
--- a/node_modules/atscm/out/lib/model/Node.js
+++ b/node_modules/atscm/out/lib/model/Node.js
@@ -5,9 +5,9 @@ Object.defineProperty(exports, "__esModule", {
 });
 exports.SourceNode = exports.ServerNode = exports.default = exports.ReferenceTypeNames = exports.ReferenceTypeIds = void 0;
 
-var _opcua_node_ids = require("node-opcua/lib/opcua_node_ids");
+var _opcua_node_ids = require("node-opcua");
 
-var _nodeclass = require("node-opcua/lib/datamodel/nodeclass");
+var _nodeclass = require("node-opcua");
 
 var _Object = require("../helpers/Object");
 
diff --git a/node_modules/atscm/out/lib/model/opcua/NodeId.js b/node_modules/atscm/out/lib/model/opcua/NodeId.js
index 5fc5e41..37cab82 100644
--- a/node_modules/atscm/out/lib/model/opcua/NodeId.js
+++ b/node_modules/atscm/out/lib/model/opcua/NodeId.js
@@ -7,7 +7,7 @@ exports.default = void 0;
 
 var _path = require("path");
 
-var _nodeid = require("node-opcua/lib/datamodel/nodeid");
+var _nodeid = require("node-opcua");
 
 /**
  * OPC-UA node id types.
diff --git a/node_modules/atscm/out/lib/model/opcua/ReverseReferenceTypeIds.js b/node_modules/atscm/out/lib/model/opcua/ReverseReferenceTypeIds.js
index 563edd7..7fc0de7 100644
--- a/node_modules/atscm/out/lib/model/opcua/ReverseReferenceTypeIds.js
+++ b/node_modules/atscm/out/lib/model/opcua/ReverseReferenceTypeIds.js
@@ -5,7 +5,7 @@ Object.defineProperty(exports, "__esModule", {
 });
 exports.default = void 0;
 
-var _opcua_node_ids = require("node-opcua/lib/opcua_node_ids");
+var _opcua_node_ids = require("node-opcua");
 
 var _Object = require("../../helpers/Object");
 
diff --git a/node_modules/atscm/out/lib/server/Session.js b/node_modules/atscm/out/lib/server/Session.js
index 765a65d..c48afd0 100644
--- a/node_modules/atscm/out/lib/server/Session.js
+++ b/node_modules/atscm/out/lib/server/Session.js
@@ -9,9 +9,9 @@ var _path = require("path");
 
 var _events = _interopRequireDefault(require("events"));
 
-var _opcua_status_code = require("node-opcua/lib/datamodel/opcua_status_code");
+var _opcua_status_code = require("node-opcua");
 
-var _opcua_client = require("node-opcua/lib/client/opcua_client");
+var _opcua_client = require("node-opcua");
 
 var _gulplog = _interopRequireDefault(require("gulplog"));
 
@@ -49,7 +49,8 @@ class Session {
       requestedSessionTimeout: 600000,
       keepSessionAlive: true,
       certificateFile: (0, _path.join)(__dirname, '../../../res/certificates/certificate.pem'),
-      privateKeyFile: (0, _path.join)(__dirname, '../../../res/certificates/key.pem')
+      privateKeyFile: (0, _path.join)(__dirname, '../../../res/certificates/key.pem'),
+      endpoint_must_exist: false
     });
     const endpoint = `opc.tcp://${_ProjectConfig.default.host}:${_ProjectConfig.default.port.opc}`;
     const timer = setTimeout(() => {
diff --git a/node_modules/atscm/out/lib/transform/ConfigTransformer.js b/node_modules/atscm/out/lib/transform/ConfigTransformer.js
index c9b52c4..4b0cffd 100644
--- a/node_modules/atscm/out/lib/transform/ConfigTransformer.js
+++ b/node_modules/atscm/out/lib/transform/ConfigTransformer.js
@@ -5,7 +5,7 @@ Object.defineProperty(exports, "__esModule", {
 });
 exports.default = void 0;
 
-var _variant = require("node-opcua/lib/datamodel/variant");
+var _variant = require("node-opcua");
 
 var _XMLTransformer = _interopRequireDefault(require("./XMLTransformer"));
 
diff --git a/node_modules/atscm/out/transform/DisplayTransformer.js b/node_modules/atscm/out/transform/DisplayTransformer.js
index 0d12f33..63ab6e9 100644
--- a/node_modules/atscm/out/transform/DisplayTransformer.js
+++ b/node_modules/atscm/out/transform/DisplayTransformer.js
@@ -5,7 +5,7 @@ Object.defineProperty(exports, "__esModule", {
 });
 exports.default = void 0;
 
-var _variant = require("node-opcua/lib/datamodel/variant");
+var _variant = require("node-opcua");
 
 var _gulplog = _interopRequireDefault(require("gulplog"));
 
diff --git a/node_modules/atscm/out/transform/Mapping.js b/node_modules/atscm/out/transform/Mapping.js
index 5f5e18b..5aaf398 100644
--- a/node_modules/atscm/out/transform/Mapping.js
+++ b/node_modules/atscm/out/transform/Mapping.js
@@ -9,7 +9,7 @@ var _path = require("path");
 
 var _assert = _interopRequireDefault(require("assert"));
 
-var _variant = require("node-opcua/lib/datamodel/variant");
+var _variant = require("node-opcua");
 
 var _Transformer = _interopRequireDefault(require("../lib/transform/Transformer"));
 
diff --git a/node_modules/atscm/out/transform/Newlines.js b/node_modules/atscm/out/transform/Newlines.js
index 55e7f56..780fa10 100644
--- a/node_modules/atscm/out/transform/Newlines.js
+++ b/node_modules/atscm/out/transform/Newlines.js
@@ -7,7 +7,7 @@ exports.default = void 0;
 
 var _os = require("os");
 
-var _variant = require("node-opcua/lib/datamodel/variant");
+var _variant = require("node-opcua");
 
 var _PartialTransformer = _interopRequireDefault(require("../lib/transform/PartialTransformer"));
 
diff --git a/node_modules/atscm/out/transform/ScriptTransformer.js b/node_modules/atscm/out/transform/ScriptTransformer.js
index 517ee6a..44a4da3 100644
--- a/node_modules/atscm/out/transform/ScriptTransformer.js
+++ b/node_modules/atscm/out/transform/ScriptTransformer.js
@@ -7,7 +7,7 @@ exports.DisplayScriptTransformer = exports.QuickDynamicTransformer = exports.Ser
 
 var _gulplog = _interopRequireDefault(require("gulplog"));
 
-var _variant = require("node-opcua/lib/datamodel/variant");
+var _variant = require("node-opcua");
 
 var _modifyXml = require("modify-xml");
 
